### DISCUSSION TITLE
Tesselating a non-planar linear ring

### DISCUSSION THREAD

**Discussion by benstadin** [pmp-library/pmp-library#169] [General]
The glu tesselator and libtess compute the triangles by first creating geometries perpendicular to the input. This is why the generated geometry can vastly differ in shape after triangulation, in case the input is a linear ring / polygon with sharper angles. 

How does pmp compare? Does it have the same behaviour, or does it provide somewhat more robustness in such case?

**Comment by dsieger** [comment]
I'm not really familiar with the GLU tessellator. Our function for polygon tessellation is [`pmp::triangulate()`](https://www.pmp-library.org/group__algorithms.html#ga6fb9055ec7d7d8c31e99bbfdfa57785a) which creates an area-minimizing triangulation of the input polygon. We don't create any special geometries first, so maybe that function already does what you are looking for.

**Comment by benstadin** [comment]
Thanks. This is indeed what I need.
